.github/workflows/makefile.yml

```diff
@@ -135,6 +135,31 @@ jobs:
       run: >
         source env/activate && make test
 
+  # Run ESLint for JavaScript code.
+  lint-js:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+
+    steps:
+    # Check out with additionally pulling the parent commit. (In case of Pull
+    # Requests, it's pointing to the base branch's head.)
+    - name: Check out sources
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+
+    - name: Install dependencies
+      uses: ./.github/actions/build-dependencies-installation
+      with:
+        toolchain: eslint
+
+    # Run the linter on suitable files. For PR's, only look at the diff against
+    # the parent commit (which, as Github executes the action on the merge
+    # commit, is the base branch). Otherwise, all files are linted.
+    - name: Run ESLint
+      run: >
+        scripts/eslint.py --base="${{ github.event_name == 'pull_request' && 'HEAD^1' || 'none' }}"
```

Comment:
Hmm, it just occurred to me that this means that when a PR modifies the linter config, we won't essentially check anything in it. If there are existing JS files that violate newly enabled checks, the PR author won't learn about them (until the PR is merged and the post-merge CI waterfall runs the linter on all files).

Either we need to detect when a PR affects the linting stuff or maybe simply run the linter on all files?..